### ISSUE TITLE
Stuck on "Preparing to back up"

### ISSUE THREAD

**Issue by raveli** [jlesage/docker-crashplan-pro#420]
Backups to local destinations work just fine but remote backups have been stuck in "preparing to back up" state for days now.

I've spent an embarrassing amount of time troubleshooting this and tried everything I can think of:

- deauthorizing the device
- restarting the container
- restarting the NAS
- created the stack + container from scratch with a new volume and just a few files in the backup set (the same files back up to a local destination successfully)
- connectivity tests (the netcat steps in Crashplan's support site) go through without any issues
- this should not be caused by file permissions etc. as the backup sets to local destinations work without errors

I do see the following error regarding an expired certificate in /config/log/service.log.0 that sounds like it could be related:

```
[04.08.23 18:49:01.285 INFO  main         42.manifest.ManifestJarValidator] Retrieved certificate chain: length=3
[04.08.23 18:49:01.294 INFO  main         42.manifest.ManifestJarValidator] Validated certificate as signed by Code42 Software[04.08.23 18:49:01.988 ERROR main         42.manifest.ManifestJarValidator] Failed to validate certificate chain, java.security.cert.CertPathValidatorException: validity chec
k failed
STACKTRACE:: java.security.cert.CertPathValidatorException: validity check failed        at java.base/sun.security.provider.certpath.PKIXMasterCertPathValidator.validate(Unknown Source)
        at java.base/sun.security.provider.certpath.PKIXCertPathValidator.validate(Unknown Source)        at java.base/sun.security.provider.certpath.PKIXCertPathValidator.validate(Unknown Source)
        at java.base/sun.security.provider.certpath.PKIXCertPathValidator.engineValidate(Unknown Source)
        at java.base/java.security.cert.CertPathValidator.validate(Unknown Source)
        at com.code42.manifest.ManifestJarValidator.validateCertChain(ManifestJarValidator.java:244)        at com.code42.manifest.ManifestJarValidator.readSignedManifestJson(ManifestJarValidator.java:145)
        at com.code42.manifest.ManifestJarValidator.execute(ManifestJarValidator.java:89)        at com.code42.manifest.InstallationValidator.doValidateInstallation(InstallationValidator.java:109)
        at com.code42.manifest.InstallationValidator.execute(InstallationValidator.java:90)        at com.backup42.service.ManifestValidationWrapper.validate(ManifestValidationWrapper.java:105)
        at com.backup42.service.CPService.start(CPService.java:359)
        at com.backup42.service.CPService.main(CPService.java:1572)
Caused by: java.security.cert.CertificateExpiredException: NotAfter: Wed Mar 01 14:00:00 EET 2023
        at java.base/sun.security.x509.CertificateValidity.valid(Unknown Source)
        at java.base/sun.security.x509.X509CertImpl.checkValidity(Unknown Source)
        at java.base/sun.security.provider.certpath.BasicChecker.verifyValidity(Unknown Source)
        at java.base/sun.security.provider.certpath.BasicChecker.check(Unknown Source)
        ... 13 more
```

I'm running the latest image `sha256:2c3f95edc657c0ade0ec1a2891ec30cdbb7e33b77e255fb9022017b579cd47c8` on QNAP NAS with Portainer using the following docker-compose file:

```
version: '3'
services:
  crashplan-pro:
    image: jlesage/crashplan-pro
    environment:
      - VNC_PASSWORD
      - CRASHPLAN_SRV_MAX_MEM
      - USER_ID
      - GROUP_ID
      - TZ=Europe/Helsinki      
    ports:
      - "5800:5800"
    volumes:
      - crashplan-config:/config
      - "/share:/storage:ro"
volumes:
  crashplan-config:
```

env:

```
- CRASHPLAN_SRV_MAX_MEM: 3GB
```

Any tips what how to troubleshoot further or fix the issue?

**Comment by jlesage** [comment]
Did you try with a minimal backup selection ?

**Comment by raveli** [comment]
Yes, I created the container and its config from scratch and only added a few files worth of approx. 10MB to backup, but it gets stuck no matter how small the backup set is.

Backing the same files to a local destination (to my PC also running Crashplan) works fine. And backups from my PC to Crashplan PRO Online are running without issues. It's only the backups from the container that are stuck in the "Preparing to back up" state after scanning the files.

**Comment by jlesage** [comment]
Is the date inside the container correct ?
```
docker exec <container name> date
```

**Comment by raveli** [comment]
I don't believe it was that, I think I checked that the time was correct as soon as I saw a mention of certificates not being valid. 

But something changed after restarting the NAS again. This time, the backups started rolling pretty much immediately after starting the container. For some reason, nothing seemed to fix the issue for so many attempts and for so many days, but suddenly, the problem was just gone and has been ever since. 

Unless you figure out anything to check, I think this issue can be closed.
